### ISSUE TITLE
No option to control the mobile data when WiFi connected

### ISSUE THREAD

**Issue by oloukanov** [GrapheneOS/os-issue-tracker#3129]
There is no option to control whether the Mobile data stays on/off when WiFi is successfully connected.

I mean a switch equivalent of the **Developer options | Mobile data always active** or included in the **Adaptive connectivity** settings the in Pixel stock OS.

**Comment by empratyush** [comment]
"Developer options | Mobile data always active" and "Adaptive connectivity" is different feature.
"Adaptive connectivity" is pixel feature [see](https://support.google.com/product-documentation/answer/12321563) .

**Comment by oloukanov** [comment]
> "Developer options | Mobile data always active" and "Adaptive connectivity" is different feature.

I agree they are different, that's why I said "**included**" in the Adaptive connectivity.
You can test if on stock after enabling Adaptive connectivity, the data connection will **not** remain connected after a successful WiFi connection. I'm talking about the data connection/disconnection effect only.

**Comment by oloukanov** [comment]
Thank you for your time addressing my question/concern.
I completely agree with the 1st part of your statement, but my observations are different from:
> It won't save any power to remain connected to cellular

**Comment by empratyush** [comment]
> I agree they are different, that's why I said "included" in the Adaptive connectivity.


"Adaptive connectivity" feature is provided by google's proprietary app.

**Comment by oloukanov** [comment]
> "Adaptive connectivity" feature is provided by google's proprietary app.
Of course. Haven't I mentioned "Stock OS" multiple times?

OK, let me try to explain it again.

**Prerequisites:**
- active data plan
- data connection enabled
- WiFi - connected

**Status check** - "Mobile Network State" on GrapheneOS found in Settings | About Phone | SIM status. In stock OS should be similar, I have no access to Pixel stock OS right now.

**Desired state** - Mobile Network State = Disconnected when connected to WiFi. Which I found is the default on other brands I checked.

**How to control** the Mobile Network State = Disconnected when connected to Wifi:
- On Pixel stock OS - enable Adaptive connectivity
- On GrapheneOS - no option to trigger it

**Workaround** - control the behavior from Developer options | Mobile data always active
